edgelet/hsm-sys/azure-iot-hsm-c/CMakeLists.txt

```diff
@@ -71,6 +72,12 @@ if(MSVC)
     add_definitions(-DGB_MEASURE_MEMORY_FOR_THIS -DGB_DEBUG_ALLOC)
 else()
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -fPIC -Wall -Werror -Wextra -Wformat=2 -Wformat-security -std=c99 ")
+
+    # suppress clang compiler warnings complaining about calls to vsnprintf
+    if(CMAKE_C_COMPILER_ID MATCHES "Clang")
```

Comment:
What was the error you saw to warrant disabling this warning?

Reply:
It is an error that looks very similar to what described [here](https://stackoverflow.com/questions/20167124/vsprintf-and-vsnprintf-wformat-nonliteral-warning-on-clang-5-0).

Reply:
Its annoying I know. I remember fixing this in some places by annotating the functions with the specified attributes as described in the article. I wonder if some new code got in that brought this warning back.